### PR TITLE
Local download

### PR DESCRIPTION
**Pull request by tonk**
In our site only the Ansible Control node is allowed to contact the Checkmk server on port 443. So the download of the agent
and the API calls need to go through the Ansible control node. 

I have added an `checkmk_agent_delegate_download` option to delegate the download.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [x] Other (please describe):
   For Red Hat and Suse based systems the agent was installed multiple times. Removed this duplicate code.

## What is the current behavior?

It is now possible to let the Ansible Control node download the Checkmk agent and transfer it to the remote node.

## What is the new behavior?

- Download the Checkmk agent on the control and transfer it when needed

### CODE REVIEW

File: roles/agent/README.md [hunk 1]
```diff
@@ -95,6 +95,10 @@ Enable this to automatically install `xinetd` on hosts with systemd prior to ver
 
 Configure the host to which Checkmk API calls are delegated to.
 
+    checkmk_agent_delegate_download: "{{ inventory_hostname }}"
+
+Configure the host to which Checkmk API dowloads are delegated to. After download the files are transfered to the remote node.
```
Comment (robin-tribe29) [inline roles/agent/README.md:100]: You might want to clarify, that this only applies, if the remote node is not downloading on its own.

File: roles/agent/tasks/main.yml [hunk 2]
```diff
@@ -22,12 +22,24 @@
...
     dest: "{{ checkmk_agent_agent.file.cre }}"
     mode: 0640
+  delegate_to: "{{ checkmk_agent_delegate_download }}"
   become: false
   when: checkmk_agent_edition == "cre"
   retries: 3
   tags:
     - download-package
 
+- name: "Transfer Checkmk CRE client to remote node"
```
Comment (robin-tribe29) [inline roles/agent/tasks/main.yml:32]: Please refer to the agent as "agent", not "client". That might be confusing and would be inconsistent.

**Comment by tonk** [comment]
Thanks for the very quick response.
I totally agree with your suggestions and have changed the code and documentation accordingly.

File: roles/agent/README.md [hunk 3]
```diff
@@ -95,6 +95,10 @@ Enable this to automatically install `xinetd` on hosts with systemd prior to ver
 
 Configure the host to which Checkmk API calls are delegated to.
 
+    checkmk_agent_delegate_download: "{{ inventory_hostname }}"
+
+Configure the host to which Checkmk API dowloads are delegated to. After download the files are transfered to the remote node, when the remote node didn't do the download.
```
Comment (robin-tribe29) [inline roles/agent/README.md:100]: Minor typo: dowloads

**Review by robin-tribe29 — CHANGES_REQUESTED** [review]
Fix the typo and we are good to merge. :)

**Comment by robin-tribe29** [comment]
The reason for the failing tests was fixed in #6b0fd6dae4d6972566a615fd30084088e85a5232, so I am going to merge this after the requested fix.

**Comment by tonk** [comment]
I fixed the `dowload` typo. No problem to merge it later. I'm already very happy that it's accepted.

**Pull request merged**